diff --git a/node_modules/@react-native/codegen/lib/parsers/typescript/components/commands.js b/node_modules/@react-native/codegen/lib/parsers/typescript/components/commands.js
index 8b81856..acc7b39 100644
--- a/node_modules/@react-native/codegen/lib/parsers/typescript/components/commands.js
+++ b/node_modules/@react-native/codegen/lib/parsers/typescript/components/commands.js
@@ -26,10 +26,14 @@ function buildCommandSchemaInternal(name, optional, parameters, types, parser) {
         null || _firstParam$typeAnnot === void 0
         ? void 0
         : _firstParam$typeAnnot.name) === 'React' &&
-      ((_firstParam$typeAnnot2 = firstParam.typeAnnotation.typeName.right) ===
+      (((_firstParam$typeAnnot2 = firstParam.typeAnnotation.typeName.right) ===
         null || _firstParam$typeAnnot2 === void 0
         ? void 0
-        : _firstParam$typeAnnot2.name) === 'ElementRef'
+        : _firstParam$typeAnnot2.name) === 'ElementRef' ||
+        ((_firstParam$typeAnnot2 = firstParam.typeAnnotation.typeName.right) ===
+        null || _firstParam$typeAnnot2 === void 0
+        ? void 0
+        : _firstParam$typeAnnot2.name) === 'ComponentRef')
     )
   ) {
     throw new Error(
